test/compilerlib/CompilerLibTest.cpp

```diff
@@ -11,95 +12,62 @@ using namespace onnx_mlir;
 
 std::string testFileName;
 std::string outputBaseName;
-std::string mcpu;
-std::string march;
-std::string mtriple;
 bool compileFromFile = false;
-bool optO0 = false;
-bool optO1 = false;
-bool optO2 = false;
-bool optO3 = false;
 
+#define IGNORE_ARG(FLAG)                                                       \
+  if (arg.find(FLAG) == 0) {                                                   \
+    return false;                                                              \
+  }
 #define PARSE_ARG(NAME, FLAG)                                                  \
   if (arg.find(FLAG) == 0) {                                                   \
     NAME = arg.substr(sizeof(FLAG));                                           \
-    return;                                                                    \
+    return true;                                                               \
   }
-
 #define PARSE_FLAG(NAME, FLAG)                                                 \
   if (arg.find(FLAG) == 0) {                                                   \
     NAME = true;                                                               \
-    return;                                                                    \
+    return true;                                                               \
   }
 
-void readArg(const std::string &arg) {
-  PARSE_ARG(mcpu, "--mcpu=");
-  PARSE_ARG(march, "--march=");
-  PARSE_ARG(mtriple, "--mtriple=");
+// Return 1 if arg used, 0 if unused.
+bool readArg(const std::string &arg) {
   PARSE_ARG(outputBaseName, "-o");
-  PARSE_FLAG(optO0, "-O0");
-  PARSE_FLAG(optO1, "-O1");
-  PARSE_FLAG(optO2, "-O2");
-  PARSE_FLAG(optO3, "-O3");
   PARSE_FLAG(compileFromFile, "--fromfile");
+  IGNORE_ARG("-"); // Ignore all other options.
   testFileName = arg;
+  return true;
 }
 
-void readCommandLine(int argc, char *argv[]) {
-  for (int i = 1; i < argc; ++i) {
-    readArg(std::string(argv[i]));
+// Read the arguments used by this program, and leave in argc/argv
+// the unused arguments, which may then be processed by the
+// ONNX-MLIR compiler.
+void readCommandLineAndKeepUnused(int &argc, char *argv[]) {
+  int num = argc;
+  argc = 0;
+  argv[argc++] = argv[0]; // Keep program name.
+  for (int i = 1; i < num; ++i) {
+    if (!readArg(std::string(argv[i]))) {
+      argv[argc++] = argv[i];
+    }
   }
 }
 
 int main(int argc, char *argv[]) {
-  readCommandLine(argc, argv);
+  // Read the compiler options from env and args.
+  readCommandLineAndKeepUnused(argc, argv);
+  omSetCompilerOptionsFromArgsAndEnv(argc, argv, nullptr);
+  std::string opt(omGetCompilerOption(OptionKind::CompilerOptLevel));
+  std::cout << "Opt Level is " << opt << std::endl;
```

Comment:
leftover naked trace

Reply:
ok, I removed it though I though it might be useful to show an example of printing options.